new e2e lsp tests can bork

### What happened?


ink stderr
⎯⎯⎯⎯⎯⎯⎯ Failed Tests 2 ⎯⎯⎯⎯⎯⎯⎯


 FAIL  src/lsp/__tests__/e2e-lsp.test.ts > LSP E2E integration (P36) > checkFile + getAllDiagnostics accumulates multi-file results
 Test Files  1 failed | 445 passed | 8 skipped (454)
AssertionError: expected false to be true // Object.is equality
      Tests  2 failed | 7424 passed | 144 skipped (7570)

   Start at  16:06:24
- Expected
   Duration  375.07s (transform 13.68s, setup 7.72s, collect 240.26s, tests 303.81s, environment 107ms, prepare 44.34s)
+ Received


- true
+ false

 ❯ src/lsp/__tests__/e2e-lsp.test.ts:321:32
    319| 
    320|       const client = config.getLspServiceClient()!;
    321|       expect(client.isAlive()).toBe(true);
       |                                ^
    322| 
    323|       await client.checkFile('/tmp/file-a.ts');

⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯[1/2]⎯

 FAIL  src/lsp/__tests__/e2e-lsp.test.ts > LSP E2E integration (P36) > MCP navigation tools are registered when navigationTools is true
AssertionError: expected 0 to be greater than 0
 ❯ src/lsp/__tests__/e2e-lsp.test.ts:469:32
    467|       (t: { serverName?: string }) => t.serverName === 'lsp-navigation…
    468|     );
    469|     expect(lspNavTools.length).toBeGreaterThan(0);
       |                                ^
    470|   });
    471| 

⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯[2/2]⎯

JUNIT report written to /Users/runner/work/llxprt-code/llxprt-code/packages/core/junit.xml
npm error Lifecycle script `test` failed with error:

npm error code 1
> @vybestack/llxprt-code@0.9.0 test
npm error path /Users/runner/work/llxprt-code/llxprt-code/packages/core
> vitest run
npm error workspace @vybestack/llxprt-code-core@0.9.0

npm error location /Users/runner/work/llxprt-code/llxprt-code/packages/core
npm error command failed
npm error command sh -c vitest run

### What did you expect to happen?

not bork

### Client information

n/a

### Login information

_No response_

### Anything else we need to know?

I think there was a corde rabbit comment on the original pr and the agent dismissed it. virtually positive its a timer thing because it only happened on mac and this doesn't look platform specific

--- a/packages/core/src/lsp/__tests__/e2e-lsp.test.ts
+++ b/packages/core/src/lsp/__tests__/e2e-lsp.test.ts
@@ -264,6 +264,12 @@ describe('LSP E2E integration (P36)', () => {
     const checkFileSpy = vi
       .spyOn(lspServiceClientModule.LspServiceClient.prototype, 'checkFile')
       .mockResolvedValue(mockDiagnostics);
+    const startSpy = vi
+      .spyOn(lspServiceClientModule.LspServiceClient.prototype, 'start')
+      .mockResolvedValue(undefined);
+    const isAliveSpy = vi
+      .spyOn(lspServiceClientModule.LspServiceClient.prototype, 'isAlive')
+      .mockReturnValue(true);
 
     try {
       const config = new Config(
@@ -281,6 +287,8 @@ describe('LSP E2E integration (P36)', () => {
       expect(diagnostics[0].severity).toBeDefined();
     } finally {
       checkFileSpy.mockRestore();
+      startSpy.mockRestore();
+      isAliveSpy.mockRestore();
     }
   });
 
@@ -310,6 +318,12 @@ describe('LSP E2E integration (P36)', () => {
         'getAllDiagnostics',
       )
       .mockResolvedValue(mockByFile);
+    const startSpy = vi
+      .spyOn(lspServiceClientModule.LspServiceClient.prototype, 'start')
+      .mockResolvedValue(undefined);
+    const isAliveSpy = vi
+      .spyOn(lspServiceClientModule.LspServiceClient.prototype, 'isAlive')
+      .mockReturnValue(true);
 
     try {
       const config = new Config(
@@ -337,6 +351,8 @@ describe('LSP E2E integration (P36)', () => {
     } finally {
       checkFileSpy.mockRestore();
       getAllDiagnosticsSpy.mockRestore();
+      startSpy.mockRestore();
+      isAliveSpy.mockRestore();
     }
   });
 
@@ -402,12 +418,24 @@ describe('LSP E2E integration (P36)', () => {
   });
 
   it('Config enables LSP when lsp is true', async () => {
-    const config = new Config(createBaseConfigParams({ lsp: true }));
-    await config.initialize();
+    const startSpy = vi
+      .spyOn(lspServiceClientModule.LspServiceClient.prototype, 'start')
+      .mockResolvedValue(undefined);
+    const isAliveSpy = vi
+      .spyOn(lspServiceClientModule.LspServiceClient.prototype, 'isAlive')
+      .mockReturnValue(true);
 
-    expect(config.getLspConfig()).toEqual({ servers: [] });
-    expect(config.getLspServiceClient()).toBeDefined();
-    expect(config.getLspServiceClient()!.isAlive()).toBe(true);
+    try {
+      const config = new Config(createBaseConfigParams({ lsp: true }));
+      await config.initialize();
+
+      expect(config.getLspConfig()).toEqual({ servers: [] });
+      expect(config.getLspServiceClient()).toBeDefined();
+      expect(config.getLspServiceClient()!.isAlive()).toBe(true);
+    } finally {
+      startSpy.mockRestore();
+      isAliveSpy.mockRestore();
+    }
   });
 
   // --- 9. Status Command ---
@@ -455,18 +483,41 @@ describe('LSP E2E integration (P36)', () => {
 
   // --- 11. Navigation Tools Registered ---
   it('MCP navigation tools are registered when navigationTools is true', async () => {
-    const config = new Config(
-      createBaseConfigParams({
-        lsp: { servers: [], navigationTools: true },
-      }),
-    );
-    await config.initialize();
+    const { PassThrough } = await import('node:stream');
+    const startSpy = vi
+      .spyOn(lspServiceClientModule.LspServiceClient.prototype, 'start')
+      .mockResolvedValue(undefined);
+    const isAliveSpy = vi
+      .spyOn(lspServiceClientModule.LspServiceClient.prototype, 'isAlive')
+      .mockReturnValue(true);
+    const getMcpTransportStreamsSpy = vi
+      .spyOn(
+        lspServiceClientModule.LspServiceClient.prototype,
+        'getMcpTransportStreams',
+      )
+      .mockReturnValue({
+        readable: new PassThrough(),
+        writable: new PassThrough(),
+      });
 
-    const tools = config.getToolRegistry().getAllTools();
-    const lspNavTools = tools.filter(
-      (t: { serverName?: string }) => t.serverName === 'lsp-navigation',
-    );
-    expect(lspNavTools.length).toBeGreaterThan(0);
+    try {
+      const config = new Config(
+        createBaseConfigParams({
+          lsp: { servers: [], navigationTools: true },
+        }),
+      );
+      await config.initialize();
+
+      const tools = config.getToolRegistry().getAllTools();
+      const lspNavTools = tools.filter(
+        (t: { serverName?: string }) => t.serverName === 'lsp-navigation',
+      );
+      expect(lspNavTools.length).toBeGreaterThan(0);
+    } finally {
+      startSpy.mockRestore();
+      isAliveSpy.mockRestore();
+      getMcpTransportStreamsSpy.mockRestore();
+    }
   });
 
   // --- 12. Navigation Tools Disabled ---
@@ -487,28 +538,51 @@ describe('LSP E2E integration (P36)', () => {
 
   // --- 13. Shutdown Lifecycle ---
   it('shutdownLspService clears client and navigation tools', async () => {
-    const config = new Config(
-      createBaseConfigParams({
-        lsp: { servers: [], navigationTools: true },
-      }),
-    );
-    await config.initialize();
+    const { PassThrough } = await import('node:stream');
+    const startSpy = vi
+      .spyOn(lspServiceClientModule.LspServiceClient.prototype, 'start')
+      .mockResolvedValue(undefined);
+    const isAliveSpy = vi
+      .spyOn(lspServiceClientModule.LspServiceClient.prototype, 'isAlive')
+      .mockReturnValue(true);
+    const getMcpTransportStreamsSpy = vi
+      .spyOn(
+        lspServiceClientModule.LspServiceClient.prototype,
+        'getMcpTransportStreams',
+      )
+      .mockReturnValue({
+        readable: new PassThrough(),
+        writable: new PassThrough(),
+      });
 
-    expect(config.getLspServiceClient()).toBeDefined();
-    const toolsBefore = config.getToolRegistry().getAllTools();
-    const navBefore = toolsBefore.filter(
-      (t: { serverName?: string }) => t.serverName === 'lsp-navigation',
-    );
-    expect(navBefore.length).toBeGreaterThan(0);
+    try {
+      const config = new Config(
+        createBaseConfigParams({
+          lsp: { servers: [], navigationTools: true },
+        }),
+      );
+      await config.initialize();
 
-    await config.shutdownLspService();
+      expect(config.getLspServiceClient()).toBeDefined();
+      const toolsBefore = config.getToolRegistry().getAllTools();
+      const navBefore = toolsBefore.filter(
+        (t: { serverName?: string }) => t.serverName === 'lsp-navigation',
+      );
+      expect(navBefore.length).toBeGreaterThan(0);
 
-    expect(config.getLspServiceClient()).toBeUndefined();
-    const toolsAfter = config.getToolRegistry().getAllTools();
-    const navAfter = toolsAfter.filter(
-      (t: { serverName?: string }) => t.serverName === 'lsp-navigation',
-    );
-    expect(navAfter).toHaveLength(0);
+      await config.shutdownLspService();
+
+      expect(config.getLspServiceClient()).toBeUndefined();
+      const toolsAfter = config.getToolRegistry().getAllTools();
+      const navAfter = toolsAfter.filter(
+        (t: { serverName?: string }) => t.serverName === 'lsp-navigation',
+      );
+      expect(navAfter).toHaveLength(0);
+    } finally {
+      startSpy.mockRestore();
+      isAliveSpy.mockRestore();
+      getMcpTransportStreamsSpy.mockRestore();
+    }
   });
 
   // --- 14. Type Contract ---

